L1Trigger/Phase2L1ParticleFlow/src/regionizer/multififo_regionizer_ref.cpp

```diff
@@ -0,0 +1,555 @@
+#include "L1Trigger/Phase2L1ParticleFlow/interface/regionizer/multififo_regionizer_ref.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/egamma/pfeginput_ref.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/regionizer/multififo_regionizer_elements_ref.icc"
+
+#include <iostream>
+#include <memory>
+
+#ifdef CMSSW_GIT_HASH
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+l1ct::MultififoRegionizerEmulator::MultififoRegionizerEmulator(const edm::ParameterSet& iConfig)
+    : MultififoRegionizerEmulator(iConfig.getParameter<uint32_t>("nEndcaps"),
+                                  iConfig.getParameter<uint32_t>("nClocks"),
+                                  iConfig.getParameter<uint32_t>("nTrack"),
+                                  iConfig.getParameter<uint32_t>("nCalo"),
+                                  iConfig.getParameter<uint32_t>("nEmCalo"),
+                                  iConfig.getParameter<uint32_t>("nMu"),
+                                  /*streaming=*/false,
+                                  /*outii=*/1,
+                                  iConfig.getParameter<bool>("useAlsoVtxCoords")) {
+  debug_ = iConfig.getUntrackedParameter<bool>("debug", false);
+  if (iConfig.existsAs<edm::ParameterSet>("egInterceptMode")) {
+    const auto& emSelCfg = iConfig.getParameter<edm::ParameterSet>("egInterceptMode");
+    setEgInterceptMode(emSelCfg.getParameter<bool>("afterFifo"), emSelCfg);
+  }
+}
+#endif
+
+l1ct::MultififoRegionizerEmulator::MultififoRegionizerEmulator(unsigned int nendcaps,
+                                                               unsigned int nclocks,
+                                                               unsigned int ntk,
+                                                               unsigned int ncalo,
+                                                               unsigned int nem,
+                                                               unsigned int nmu,
+                                                               bool streaming,
+                                                               unsigned int outii,
+                                                               bool useAlsoVtxCoords)
+    : RegionizerEmulator(useAlsoVtxCoords),
+      NTK_SECTORS(9),
+      NCALO_SECTORS(3),
+      NTK_LINKS(2),
+      NCALO_LINKS(2),
+      HCAL_LINKS(0),
+      ECAL_LINKS(0),
+      NMU_LINKS(1),
```

Comment:
Same as above for `TrackInputEmulator`, it seems to me that there are some of these that are actually constants across constructors (e.g. `NMU_LINKS` is always `1`). Do you plan to have them configurable in future?

Reply:
@gpetruc Can you answer this question?